### PR TITLE
Task5 hooks

### PR DESCRIPTION
**Pull request by SteveRusin**

### CODE REVIEW

File: src/home-page/Content/Movies/MoviesWrapper.tsx [hunk 1]
```diff
@@ -1,23 +1,72 @@
...
-    <Wrapper>
-      {movies.map((movie) => (
-        <Movie movie={movie} key={movie.id} />
-      ))}
-    </Wrapper>
+    <>
+      <Wrapper>
+        {movies.map((movie) => (
+          <Movie
+            onMenuItemClick={(id) => {
```
Comment (ygorbunkov) [inline src/home-page/Content/Movies/MoviesWrapper.tsx:32]: I would say this chunk of js doesn't fit too nicely into JSX-block, so I would rather defined that as a `const`

File: src/home-page/Header/Header.tsx [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
 import { SearchWrapper } from './HeaderSearch';
-import { HeaderWrapper } from './HeaderWrapper';
+import { HeaderWrapper } from './Header.styles';
+import { MovieDetails } from './MovieDetails';
 
 export const HomePageHeader = () => {
   const [isOpen, setIsOpen] = useState(false);
+  const { movie, showMovieDetails } = useContext(MovieDetailsContext);
+
+  const hideMovieDetails = useCallback(() => showMovieDetails(null), []);
```
Comment (ygorbunkov) [inline src/home-page/Header/Header.tsx:19]: I may be missing something, but these event handlers don't seem to depend on anything _external_, so they will trigger state update and consequent re-render, despite of referential equality. Considering `useMemo()` / `useCallback()` optimization [comes at a cost](https://kentcdodds.com/blog/usememo-and-usecallback), what's the point using them here?

Comment (SteveRusin) [reply src/home-page/Header/Header.tsx:19]: That's a top component so I declare callbacks here, so children don't get re-rendered. Anyway, it was a part of HW, I wouldn't use this hook here anyway.

File: src/home-page/Content/Movies/Movie.tsx [hunk 3]
```diff
@@ -1,22 +1,42 @@
...
-    <MovieWrapper tabIndex={0}>
-      <img className='movie-cover' src={coverUrl} alt='movie cover' />
+    <MovieWrapper tabIndex={0} onClick={() => onMovieClick(movie)}>
+      <MovieCover src={coverUrl} alt='movie cover' />
       <div className='movie-description-container'>
         <div className='movie-description-heading'>
           <span className='movie-title'>{title}</span>
           <span className='movie-genre'>{genre}</span>
         </div>
         <span className='movie-release-date'>{releaseDate}</span>
```
Comment (ygorbunkov) [inline src/home-page/Content/Movies/Movie.tsx:24]: Your release date seems to come as ISO string. I would say, something more _human readable_ (e.g. `dd/mm/yyyy`) wouldn't hurt the UX too much.

**Pull request merged**